### ISSUE TITLE
CAM no longer builds with intel-oneapi compilers

### ISSUE THREAD

**Issue by gdicker1** [EarthWorksOrg/EarthWorks#35]
@dazlich discovered that the update to include ESCOMP/CAM cam6_3_148 in [EarthWorksOrg/CAM #14](https://github.com/EarthWorksOrg/CAM/pull/14) broke builds with the Intel-OneAPI compilers

Example of error message:

>`ERROR: Command: '/glade/u/home/dazlich/ewv2.1/post_relv21/components/cam/bld/configure -s -fc_type intel-oneapi -dyn mpas -hgrid mpasa120 -cpl nuopc -usr_src /glade/derecho/scratch/dazlich/frametest.intel-oneapi/SourceMods/src.cam -spmd -nosmp -ocn mpaso -dyn mpas -nlev=32 -mpas_libdir /glade/derecho/scratch/dazlich/frametest.intel-oneapi/bld/atm/obj/mpas' failed with error 'ERROR: intel-oneapi is not a valid value for parameter fc_type: valid values are cray,pgi,intel,gnu,pathscale,ibm,nag,nvhpc,pgi-gpu,nvhpc-gpu' from dir '/glade/derecho/scratch/dazlich/frametest.intel-oneapi/Buildconf/camconf'`

This seems to be due to edits to bld/configure (near line 1685) that were introduced in the commit [14b6813](https://github.com/ESCOMP/CAM/commit/14b6813e931f7a4622c5175ca007d9c371ea76d7) in ESCOMP/CAM as part of the changes in cam6_3_147. I think this could be fixed by either reverting this change or replacing the logic with correct Perl substring matching.

**Comment by gdicker1** [comment]
Example fix from [this StackOverflow comment](https://stackoverflow.com/a/7283308):

```perl
if (index($str, $substr) != -1) {
    print "$str contains $substr\n";
} 
```

Instead of 
```perl
if ($fc_type eq "oneapi") {$fc_type = 'intel'; }
```

**Comment by dazlich** [comment]
Jim Edwards just had a look and concurs with the example fix.

On Mar 8, 2024, at 1:10 PM, G. Dylan Dickerson ***@***.***> wrote:


Example fix from this StackOverflow comment<https://stackoverflow.com/a/7283308>:

if (index($str, $substr) != -1) {
    print "$str contains $substr\n";
}

Instead of

if ($fc_type eq "oneapi") {$fc_type = 'intel'; }

**Comment by gdicker1** [comment]
@dazlich, nice! Do you know if he's opening an issue about it in ESCOMP/CAM, or should I?

**Comment by dazlich** [comment]
This is from a meeting chat we are currently in:

Jim Edwards
1:27 PM
if ($fc_type =~ /oneapi/) fc_type = 'intel’;

Jim says he’ll take care of it.


On Mar 8, 2024, at 1:29 PM, G. Dylan Dickerson ***@***.***> wrote:


** Caution: EXTERNAL Sender **


Jim Edwards just had a look and concurs with the example fix.

@dazlich<https://github.com/dazlich>, nice! Do you know if he's opening an issue about it in ESCOMP/CAM, or should I?

**Comment by gdicker1** [comment]
Gotcha, thanks Don.

**Comment by gdicker1** [comment]
Closed by #32
